Update the README

ppraegla opened the pull request:
## Description and Context
Update the `README.md`. I was inspired by the README of [Queens](https://github.com/queens-py/queens).

I also removed the complete setup process of 4C from the README. It was too long, and it duplicates the information from our documentation.

I also moved the setup process to the top level and named it 'Installation'. Technically, it is not an installation of 4C. But I think everyone will understand that this is the page where you find information on how to set up 4C. 

## Related Issues and Pull Requests
Closes #279

ppraegla commented on README.md:
````diff
@@ -25,136 +25,33 @@
-ninja -j <numProcs> full
+To quickly run 4C, you can use our docker image, where 4C comes pre-compiled.
 ```
-
-#### Run the Tests
````
I'm not too happy about docker. It makes it very easy to use 4C. However, it is pretty cumbersome to get the simulation results out of the docker container.

ppraegla commented on doc/readthedocs/index.rst:
```diff
@@ -38,6 +38,9 @@ This guide to |FOURC| is structured as follows:
 :ref:`Part II - The 4C Community<4Ccommunity>`
    A brief summary of the roles and responsibilities within the |FOURC| community.
 
+:ref:`Installation<Installation>`
+   A summary of all requirements of |FOURC| and detailed steps how to build |FOURC|.
+
 :ref:`Part III — Tutorials<tutorials>`
    A series of beginner-level tutorials showcases the setup procedure for specific application scenarios.
```
I did not update the `Part` section because I think this should be rather removed as updating the list means updating the different parts.

ppraegla commented on README.md:
```diff
@@ -25,136 +25,33 @@
 to address a plethora of physical problems by means of _computational mechanics_.
 
 Large parts of 4C are based on finite element methods (FEM),
-but alternative discretization methods such as discontinuous Galerkin methods (DG),
+but alternative discretization methods, such as discontinuous Galerkin methods (DG),
 particle methods and mesh-free methods have also been successfully integrated.
 The research software is implemented throughout in object-oriented programming (C++)
 using modern software design and is parallelized with MPI for distributed memory hardware architectures.
```
We could also add some capabilities here, like on the website.

ppraegla commented on README.md:
````diff
@@ -25,136 +25,33 @@
-
-```bash
-<path/to/build/dir>/4C <jobName>.dat <outputName>
+docker run --interactive --tty ghcr.io/4c-multiphysics/4c:main
+/home/user/4C/build/4C ../tests/input_files/<some-input-file>.4C.yaml output_name
````
I'm not too happy about docker. It makes it very easy to use 4C. However, it is pretty cumbersome to get the simulation results out of the docker container.

mayrmt commented on README.md:
```diff
@@ -25,136 +25,33 @@
-by [creating a GitHub issue](https://github.com/4C-multiphysics/4C/issues). Use an issue template to pre-populate the *Description* field, giving you instructions on submitting the issue.
-
-For more general questions, you can reach us on the [4C Slack workspace](https://join.slack.com/t/4c-multiphysics/shared_invite/zt-1oi61jgdd-5tZuHku3Tb_BH5UBgojbpQ).
+See our [documentation](https://4c-multiphysics.github.io/4C/readthedocs/installation.html) for more options on how to set up 4C.
```
```suggestion
See our [documentation](https://4c-multiphysics.github.io/4C/readthedocs/installation.html) for more information on how to build 4C from source on your machine.
```

gilrrei commented:
I realized that 4C, as well as QUEENS, put the links to the documentation and website into the badges. This is not common in other projects on github, such that QUEENS ended up changing it.

mayrmt commented:
I like the links in the badges, because that's a prominent and easy-to-find placement.

mayrmt commented:
Maintainers have discussed this: let's keep the badges as they are.

ppraegla commented:
Please re-review, I added the suggestions. I also added links to the website and documentation additionally to the badges. I think, this makes the documentation more prominent. 

@ischeider Please have a look at the changes to the documentation.

sebproell commented on doc/readthedocs/installation.rst:
```diff
@@ -38,7 +33,7 @@ So we have to install TPLs first. Here's an list of the |FOURC|-related tools an
 
 - git
-- c++ compiler with c++17 compatibility (e.g. gcc 11)
+- c++ compiler with c++20 compatibility (e.g. gcc 11)
```
```suggestion
- c++ compiler with c++20 compatibility (e.g. gcc 13)
```

sebproell reviewed (commented):
Looks good!

mayrmt commented on doc/readthedocs/developer_cluster.rst:
```diff
@@ -7,7 +7,7 @@ In general there are several ways to get the |FOURC| source code to a cluster. I
 - Limitations due to firewall settings may emerge.
 - In case you want to commit changes in your repository on a cluster,
-  remember to set up |FOURC| as described in the |FOURC| ``README.md`` or in the :ref:`Setup guide <SetupGuideto4C>`.
+  remember to set up |FOURC| as described in the :ref:`Installation guide <installation>`.
```
At second thought, one could argue that "installation" is not an optimal word here, since one usually does not `make install` 4C, but just `make` (build) it. On the other hand, "installation guide" conveys the proper meaning of "getting everything up and running".

I am fine with the current wording. Just wanted to raise awareness.

mayrmt commented on README.md:
```diff
@@ -25,136 +25,37 @@
-
-For more general questions, you can reach us on the [4C Slack workspace](https://join.slack.com/t/4c-multiphysics/shared_invite/zt-1oi61jgdd-5tZuHku3Tb_BH5UBgojbpQ).
+See our [documentation](https://4c-multiphysics.github.io/4C/readthedocs/installation.html) for more information on how to build 4C from source on your machine.
+Condult our [Tutorials](https://4c-multiphysics.github.io/4C/readthedocs/tutorials.html) to get an overview of the general workflow in 4C.
```
```suggestion
Consult our [Tutorials](https://4c-multiphysics.github.io/4C/readthedocs/tutorials.html) to get an overview of the general workflow in 4C.
```

ppraegla replied:
I thought, I just accept your suggestion to avoid any mistakes from copy-paste. I should have read the suggestion better :sweat_smile:

mayrmt reviewed (changes requested):
Just a typo to be fixed before merging...
